src/database/mod.rs

```diff
@@ -325,6 +327,74 @@ impl Database {
         let result = self.pool.fetch_one(query).await?;
         Ok(result.get::<i64, _>(0) as i32)
     }
+
+    pub async fn get_provers(&self) -> Result<prover::Provers, Error> {
+        let query = sqlx::query(
+            r#"
+                SELECT batch_size, url, timeout_s
+                FROM provers
+            "#,
+        );
+
+        let result = self.pool.fetch_all(query).await?;
+
+        Ok(result
+            .iter()
+            .map(|row| {
+                let batch_size = row.get::<i64, _>(0);
+                let url = row.get::<String, _>(1);
+                let timeout_s = row.get::<i64, _>(2);
+                prover::Prover {
+                    url,
+                    batch_size,
+                    timeout_s,
+                }
+            })
+            .collect::<prover::Provers>())
+    }
+
+    pub async fn write_prover(
+        &self,
+        batch_size: usize,
+        url: impl ToString,
+        timeout_seconds: u64,
+    ) -> Result<(), Error> {
+        let mut tx = self.pool.begin().await?;
+        let url = url.to_string();
+
+        let query = sqlx::query(
+            r#"
+                INSERT INTO provers (batch_size, url, timeout_s)
+                VALUES ($1, $2, $3)
+                ON CONFLICT (batch_size)
+                DO UPDATE SET (url, timeout_s) = ($2, $3)
+            "#,
+        )
+        .bind(batch_size as i64)
+        .bind(url)
+        .bind(timeout_seconds as i64);
+
+        tx.execute(query).await?;
+        tx.commit().await?;
+
+        Ok(())
+    }
+
+    pub async fn remove_prover(&self, batch_size: usize) -> Result<(), Error> {
+        let mut tx = self.pool.begin().await?;
+
+        let query = sqlx::query(
+            r#"
+              DELETE FROM provers WHERE batch_size = $1  
+            "#,
+        )
+        .bind(batch_size as i64);
+
+        tx.execute(query).await?;
```

Comment:
nit: do we need an explicit transaction here? wouldn't `self.pool.execute(query)` work the same?